che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -933,26 +940,12 @@ public PushResponse push(PushRequest request) throws GitException, UnauthorizedE
             }
 
             Iterable<PushResult> list = (Iterable<PushResult>)executeRemoteCommand(remoteUri, pushCommand);
-            for (PushResult pushResult : list) {
-                Collection<RemoteRefUpdate> refUpdates = pushResult.getRemoteUpdates();
-                for (RemoteRefUpdate remoteRefUpdate : refUpdates) {
-                    if (!remoteRefUpdate.getStatus().equals(org.eclipse.jgit.transport.RemoteRefUpdate.Status.OK)) {
-                        if (remoteRefUpdate.getStatus().equals(org.eclipse.jgit.transport.RemoteRefUpdate.Status.UP_TO_DATE)) {
-                            return newDto(PushResponse.class).withCommandOutput("Everything up-to-date");
-                        } else {
-                            throw new GitException(remoteRefUpdate.getStatus().toString());
-                        }
-                    }
-                }
-            }
+            pushResult = list.iterator().next();
         } catch (GitAPIException exception) {
-            if ("origin: not found.".equals(exception.getMessage())) {
-                throw new GitException(ERROR_NO_REMOTE_REPOSITORY, exception);
-            } else {
-                throw new GitException(exception.getMessage(), exception);
-            }
+            PushResponse jGitPushResponse = new JGitPushResponse(true);
+            return jGitPushResponse.withCommandOutput(exception.getMessage());
         }
-        return newDto(PushResponse.class).withCommandOutput("Successfully pushed to " + remoteUri);
+        return new JGitPushResponse(request, pushResult, currentBranch);
```

Comment:
IMHO It would be better not to wrap `PushResponse` DTO but create missing methods in the DTO and place the logic inside the method in `JGitConnection`.